tests/cluster/etl/test_loader.py

```diff
@@ -8,55 +8,78 @@
 
-        input = {"myset": self.df}
-        result = self.loader.etl(input)
+        self.df = create_dataframe()
+        self.df_empty = create_empty_dataframe()
 
-        # assert that loader returns something:
+    def test_return_inputs(self):
+        input = {"df": self.df}
+        result = self.loader.etl(input)
         self.assertIs(input, result)
 
-        args = self.loader.save.call_args[
-            0
-        ]  # the .args part became available in python 3.8
-        print(args)
-        self.assertEqual(len(args), 1)
-        self.assertIs(args[0], self.df)
+    def test_save(self):
+        self.loader.etl({"df": self.df})
+        self.assertTrue(self.loader.save.called)
+        self.assertFalse(self.loader.save_many.called)
 
     def test_save_many(self):
-
-        self.loader.etl({"myset1": self.df, "myset2": self.df})
-
-        args = self.loader.save_many.call_args[
-            0
-        ]  # the .args part became available in python 3.8
-        print(args)
-        self.assertEqual(len(args), 1)
-        datasets = args[0]
-        self.assertEqual(len(datasets), 2)
-        self.assertEqual(set(datasets.keys()), {"myset1", "myset2"})
+        self.loader.etl({"df1": self.df, "df2": self.df})
+        self.assertFalse(self.loader.save.called)
+        self.assertTrue(self.loader.save_many.called)
+
+    def test_save_input_key(self):
+        self.loader_input_key.etl({"df": self.df_empty, "my_input_key": self.df})
+        self.assertTrue(self.loader_input_key.save.called)
+        self.assertFalse(self.loader_input_key.save_many.called)
+        self.assertIs(self.loader_input_key.save.call_args[0][0], self.df)
+
+    def test_save_many_input_key_list(self):
+        self.loader_input_key_list.etl(
+            {"df": self.df_empty, "my_input_key_1": self.df, "my_input_key_2": self.df}
+        )
+        self.assertFalse(self.loader_input_key_list.save.called)
+        self.assertTrue(self.loader_input_key_list.save_many.called)
+
+        for input_key in self.dataset_input_key_list:
+            self.assertTrue(
+                input_key
+                in dict(self.loader_input_key_list.save_many.call_args[0][0]).keys()
+            )
 
 
 def create_dataframe():
-    return Spark.get().createDataFrame(
-        Spark.get().sparkContext.parallelize([(1, "1"), (2, "2"), (3, "3")]),
-        StructType(
-            [
-                StructField("id", IntegerType(), False),
-                StructField("text", StringType(), False),
-            ]
-        ),
+    data = [(1, "1"), (2, "2"), (3, "3")]
+
+    schema = StructType(
+        [
+            StructField("id", IntegerType(), False),
+            StructField("text", StringType(), False),
+        ]
     )
 
+    return Spark.get().createDataFrame(data=data, schema=schema)
+
+
+def create_empty_dataframe():
+    data = [()]
+    return Spark.get().createDataFrame(data=data)
+
 
 if __name__ == "__main__":
     unittest.main()
```

Comment:
why line 84 and 85?

Reply:
I ran these locally when writing the tests to check along the way. I just left it for ease in the future - should remove?

Reply:
In my opinion I would remove it.